packages/amesos2/src/Amesos2_Superludist_decl.hpp

```diff
@@ -308,11 +312,11 @@ class Superludist : public SolverCore<Amesos2::Superludist, Matrix, Vector>
 
   // The following Arrays are persisting storage arrays for A, X, and B
   /// Stores the values of the nonzero entries for SuperLU_DIST
-  Teuchos::Array<slu_type> nzvals_;
+  host_value_type_array nzvals_view_;
   /// Stores the row indices of the nonzero entries
-  Teuchos::Array<SLUD::int_t> colind_;
+  host_ordinal_type_array colind_view_;
   /// Stores the location in \c Ai_ and Aval_ that starts row j
-  Teuchos::Array<SLUD::int_t> rowptr_;
+  host_size_type_array rowptr_view_;
   /// 1D store for B values
   mutable Teuchos::Array<slu_type> bvals_;
```

Comment:
Don't the bvals_ and xvals_ need to be converted to the Kokkos::View equivalents as well?

Reply:
It looks like we still have this interface for MultiVector. For MultiVector, Tpetra is copying the numerical values from input to output vectors, while for CrsMatrix, it is doing a shallow-copy and needs to use Kokkos::View? We could double-check with team Tpetra. I'll just tag @brian-kelley, for now..

Reply:
@iyamazaki If you replaced bvals and xvals with Kokkos views as well (with the correct type), then the Amesos2 utility ``get_1d_copy_helper::do_get`` would also be able to shallow copy. Right now, with them being Arrays, it is always a deep copy. But it's hard to tell at a glance if shallow copying will cause issues with Tpetra's view reference counting.

Reply:
Thank you, @brian-kelley & @ndellingwood. I am thinking to keep what we have for now (it sounds safer and probably behaves similar to what we used to have with deprecated codes). What do you think?

Reply:
I think it's fine to leave it as is. Copying a vector is probably really cheap compared to the Superlu solve anyway.

Reply:
That's fine then to leave as is for now. Will need the `get_1d_copy_helper_kokkos_view::do_get` (also requires an extra boolean as first arg) version when updating to use Kokkos::Views for bvals_ and xvals_